apps/ai-image-generator/backend/src/actions/aiig-erase-and-fill.ts

```diff
@@ -0,0 +1,86 @@
+console.log('prod');
+import { AppActionCallContext } from '@contentful/node-apps-toolkit';
+import { PlainClientAPI } from 'contentful-management';
+import { OpenAiApiService } from '../services/openaiApiService';
+import * as nodeFetch from 'node-fetch';
+import { AppActionCallResponse, Image } from '../types';
+import { fetchOpenAiApiKey } from '../utils';
+
+interface AppActionCallParameters {
+  prompt: string;
+  image: string;
+  mask: string;
+}
+
+export interface ImageEditResult {
+  type: 'ImageEditResult';
+  images: Image[];
+}
+
+export const handler = async (
+  payload: AppActionCallParameters,
+  context: AppActionCallContext
+): Promise<AppActionCallResponse<ImageEditResult>> => {
+  const {
+    cma,
+    appActionCallContext: { appInstallationId },
+  } = context;
+
+  let images: Image[];
+
+  try {
+    const openAiApiKey = await fetchOpenAiApiKey(cma, appInstallationId);
+    const openAiApiService = OpenAiApiService.fromOpenAiApiKey(openAiApiKey);
+    const { prompt, image: imageUrl, mask: maskUrl } = payload;
+    const fetch = nodeFetch.default;
```

Comment:
This looks a bit kludgy but it's because it's super hard to mock default functions using sinon in Node 🤷